neuraxle/metrics.py

```diff
@@ -78,33 +83,35 @@ def _initialize_metrics(self, metrics):
             self.metrics_results_train[m] = []
             self.metrics_results_validation[m] = []
 
+    def _did_fit_transform(self, data_container: DataContainer, context: ExecutionContext) -> DataContainer:
+        return self._did_fit_transform_or_transform(data_container, context)
+
     def _did_transform(self, data_container: DataContainer, context: ExecutionContext) -> DataContainer:
+        return self._did_fit_transform_or_transform(data_container, context)
+
+    def _did_fit_transform_or_transform(self, data_container: DataContainer, context: ExecutionContext):
         """
-        Calculate metrics results after transform if there is an expected outputs in the data container.
+        Calculate metrics results after fit, or transform if there is an expected outputs in the data container.
+        Also, calculate validation metrics if there is a sub data container named validation in the data container.
+        Please refer to :class:`neuraxle.data_container.DataContainer` for more information about sub data containers.
 
         :param data_container: data container to calculate metrics for
         :type data_container: DataContainer
-        :param context: execution context
         :return: data container
         :rtype: DataContainer
         """
-        if data_container.expected_outputs is not None and len(data_container.expected_outputs) > 0:
-            self._calculate_metrics_results(data_container)
+        if data_container.expected_outputs is None or len(data_container.expected_outputs) == 0:
+            return data_container
 
-        return data_container
+        self._calculate_metrics_results(data_container)
```

Comment:
Would it be possible that a metric uses only the predicted and not at all any expected? 

In that case we might need to replace one `None` in expected_outputs by several `[None] * N`. 

please resolve to issue

Reply:
https://github.com/Neuraxio/Neuraxle/issues/253